src/vcf/parsing_state.cpp

```diff
@@ -87,5 +91,31 @@ namespace ebi
     {
         defined_metadata.emplace(meta_type, id);
     }
+
+    void ParsingState::validate_additional_checks()
+    {
+        if (additionalChecks.checkEvidence) {
+            if (genotypes_present() || allele_frequencies_present()) {
+                return;
+            }
+            throw new InfoBodyError{record->line, "Genotypes or Allele frequencies are required"};
```

Comment:
if there are no genotypes, that's not exactly a INFO body error. I would use a BodySectionError. Also, please explain better to the user what exact fields are supposed to be filled. 

I would just use the basic message, making the string a bit longer (but concise!). It doesn't apply to this error, but for other errors we use the detailed message, and the difference between the basic message and the detailed message, is that we group by the basic message (exact string), and the detailed message has concrete data about the error (that will change across lines), e.g. `GL was "0,1", but expected 3 values`.